src/app/api/profiles/[id]/reject/route.ts

```diff
@@ -1,30 +1,44 @@
-import { saveRejectingReason } from '@/backend/profile/rejection.service'
+import { sendProfileRejectedEmail } from '@/actions/mailing/sendProfileStatusEmail'
+import {
+  deleteRejectingReason,
+  saveRejectingReason,
+} from '@/backend/profile/rejection.service'
 import { requireUserRoles } from '@/utils/auths'
 import { Role } from '@prisma/client'
 import { NextRequest, NextResponse } from 'next/server'
 
-export async function POST(
-  request: NextRequest,
-  { params }: { params: { id: string } },
-) {
-  const profileId = params.id
-
+export async function POST({
+  request,
+  params: { id: profileId },
+}: {
+  request: NextRequest
+  params: { id: string }
+}) {
   if (!requireUserRoles([Role.MODERATOR])) {
     return new NextResponse(null, { status: 401 })
   }
 
   try {
-    const rejectingReasonText = await request.json()
-    const createdReason = await saveRejectingReason(
-      profileId,
-      rejectingReasonText.reason,
-    )
+    const { email, reason } = await request.json()
```

Comment:
**nit:** Why passing email if you can just `findProfileById` and get the most safe way